xformers/components/attention/blocksparse.py

```diff
@@ -86,39 +85,24 @@ def __init__(
             ), "Only block sizes in [16, 32, 64] are supported"
 
             super().__init__()
+
+            self.causal = causal
+
             self.attn_drop = torch.nn.Dropout(dropout, inplace=False)
 
             # Pure blocksparse data
             self.layout = layout
             self.block_size = block_size
 
-            # blocksparse operators
-            self.sparse_dot_sdd = blocksparse_matmul(
-                self.layout,
-                self.block_size,
-                "sdd",
-                trans_a=False,
-                trans_b=True,
-            )
-            self.sparse_dot_dsd = blocksparse_matmul(
-                self.layout,
-                self.block_size,
-                "dsd",
-                trans_a=False,
-                trans_b=False,
-            )
-            self.sparse_softmax = blocksparse_softmax(self.layout, self.block_size)
-
             # make sure that the head dimension is not folded down with the batch
             self.requires_head_dimension = True
 
             # key padding mask and attention mask must be passed in separately
-            self.requires_separate_masks = True
             self.requires_same_k_q_dimensions = True
 
-            # Properties specific to this attention mechanism
-            self.supports_attention_mask = True
-            self.supports_key_padding_mask = True
+            # The underlying triton op does not support per element attention mask
+            self.supports_attention_mask = False
+            self.supports_key_padding_mask = False
 
         def update_mask_type(self, mask: torch.Tensor):
```

Comment:
Was just reading the code and realised this is not used anymore, safe to delete?

Reply:
Nice catch ! In general I need to give it a second look and clean things up, I was waiting for @colehawkins so that there's no conflict but will do

Reply:
@blefaudeux  Posted in #277. Pending CI, but tests passed locally so I have high hopes.